index.d.ts

```diff
@@ -62,5 +65,15 @@ export class KeyboardAvoidingView extends React.Component<
 > {}
 export interface TextProps extends NativeTextProps {}
 export function Text(props: NativeTextProps): NativeText;
+export interface TextInputProps extends NativeTextInputProps {
+  error?: string | null;
+  errorStyle?: StyleProp<TextStyle>;
+  optionText?: string;
+  optionTextStyle?: StyleProp<TextStyle>;
+}
+export function TextInput(
+  props: TextInputProps
+): React.Component<TextInputProps>;
+
```

Comment:
if we allow user complete control the error and option component, we can't prevent layout unstable when error message not show.

Reply:
I agree. But let's change the type of `error` and `option` to `React.ReactNode` so that the user can render `FormattedMessage` directly without imperative translation.

```typescript
interface Props extends NativeTextInputProps {
  error?: React.ReactNode; // typically it is string or something that renders to string, though we cannot have compile checking to ensure this.
  errorStyle?: StyleProp<TextStyle>;
  option?: React.ReactNode;
  optionStyle?: StyleProp<TextStyle>;
}
```

Actually what we should support is wherever we use `Text`, we should support `React.ReactNode` instead of `string`. Doing so allows us to support `FormattedMessage` `react-intl` or `react-messageformat`.

One case that we cannot support `React.ReactNode` is the underlying component is taking plain string as prop, e.g. `<TextInput placeholder="placeholder" />`. In this case, just support `string`.

Reply:
Let's review the current component API and change `string` to `React.ReactNode` whenever possible.